### ISSUE TITLE
Multi-value support for GM_getValues, GM_setValues, GM_deleteValues?

### ISSUE THREAD

**Issue by tophf** [Tampermonkey/tampermonkey#2045]
Maybe something like this?

* `GM_getValues(['foo', 'bar'])` -> {foo: 1, bar: 2} or {foo: 1} if `bar` is missing
* `GM_getValues({foo: 1, bar: 2})` -> providing the default values.
* `GM_getValues()` or `GM_getValues(null)` or `GM_getValues(undefined)` -> read the entire storage
---
* `GM_setValues({foo: 1, bar: 2})`
* `GM_deleteValues(['foo', 'bar'])`

Ideally I'd like Violentmonkey/Tampermonkey to avoid preloading all storage data at script start when the script doesn't ask for the legacy GM_xxxValue, in which case multi-value support becomes important because of the overhead to process each value individually. When `GM.xxxValue` functions are implemented as a native asynchronous call it makes sense from a performance viewpoint to allow specifying multiple values in one call, just like it's done in `chrome.storage` API.

**Comment by derjanb** [comment]
Makes sense. I'll add support for this in the next BETA version.

**Comment by cyfung1031** [comment]
Could there be something "Proxy" ?
So developers do not need to think too much about sync the values along tabs.
and don't need `getValue` and `setValue`

### Option 1
```js
// only one argument. object. the value is the default value if not exist (so this can make sure there is always an default value)
const gmValues = GM_values({
foo: 1,
bar: 2
});

// gmValues is a Proxy
// the values are obtained at each proxy's get

console.log(await gmValues.foo) // 1
gmValues.foo = 3
console.log(await gmValues.foo) // 3
```

### Option 2

```js
// only one argument. object. the value is the default value if not exist (so this can make sure there is always an default value)
const gmValues = await GM.values({
foo: 1,
bar: 2
});

// gmValues is a Proxy. All values inside will have GM_addEvenListener. 
// the values will be automatically updated, so getter just read the stored cache value.

console.log(gmValues.foo) // 1
gmValues.foo = 3
console.log(gmValues.foo) // 3
```

**Comment by tophf** [comment]
With proxy you can't await the result of setting the value. Either way, it's not related to the idea in this issue, so you should open a new one.

**Comment by erosman** [comment]
If I may add a word, FM introduced multi-value GM storage option in https://github.com/erosman/support/issues/545
Initially, using a new API but changing to an option for the existing API, which mimics the `browser.storage.local.get()`.

**Comment by tophf** [comment]
I thought about extending the existing singular functions with the new syntax, but I decided it would encourage authors to use it directly without first checking for `typeof GM.getValues === 'function'`.

**Comment by derjanb** [comment]
Fixed at 5.3.6207 ([crx](https://tampermonkey.net/crx/tampermonkey_beta_current.crx)|xpi in review)

I've decided against overloading the old API methods, because it would be difficult to detect whether multi-values are supported or not.
